leader/leader.go

```diff
@@ -154,6 +163,7 @@ func (l *Client) AddWatch(key string, retry time.Duration, valuesC chan string)
 			select {
 			case valuesC <- resp.Node.Value:
 			case <-l.closeC:
+				logger.Info("Watcher has been closed.")
 				return
 			}
 		}
```

Comment:
Sorry, commenting on some existing code, but it's potentially related to the originally reported problem. 

On Line 165, if the watcher is returned with a response, the code will enter then enter the watcher != nil on line 168, which will drop the resp and wait for the next etcd event (which might be awhile). I believe Next will block until that event, so this could be a source of missed events on the client side.

My suspicion is that it should be an else. 
```
if watcher == nil {
// re-create the watcher and process the current value
} else {
// get the next event from the watcher
}
```

Reply:
If I'm reading your line numbers correctly, you are referring to this (L175 from [here](https://github.com/gravitational/coordinate/pull/12/commits/0900c51a13de171f9c8509525cdeb4925b1477df#diff-564ef4f55f8e721c554bbbba0b9678eeR155)):
```go
if watcher == nil {
	watcher, resp, err = l.getWatchAtLatestIndex(ctx, api, key, retry)
}

if watcher != nil {
	// BUG: previous resp is dropped
	resp, err = watcher.Next(ctx)
	if err == nil {
		if resp.Node.Value == "" {
			continue
		}
		backoff.Reset()
	}
}
```

Reply:
correct, yea sorry I should have just included the code.